### PR TITLE
upgrade to NANv2

### PR DESCRIPTION
**Pull request by julianduque**
Add support for io.js 3.0.0 and the upcoming Node.js 4.0.0

### CODE REVIEW

**Comment by c4milo** [comment]
Yes! this will fix #121

**Comment by sonnyp** [comment]
@julianduque thanks!
Could you just replace "iojs-v2.5.0" with "iojs" in .travis.yml ?

**Comment by c4milo** [comment]
@sonnyp https://github.com/node-xmpp/node-expat/pull/124

**Comment by julianduque** [comment]
@c4milo @sonnyp it should use g++-4.8 on travis, check this for reference: https://github.com/justinlatimer/node-midi/commit/6b2fff5317e98abd82278ffa2470963acc14432c

**Comment by c4milo** [comment]
@julianduque @sonnyp I already upgraded .travis.yml to use g++ 4.8, it's failing right now because this PR hasn't been merged.

**Comment by julianduque** [comment]
Added the travis update in this PR, thanks @c4milo

**Comment by julianduque** [comment]
@sonnyp @c4milo done with all io.js versions

**Comment by c4milo** [comment]
:dancer:

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,25 @@
-sudo: false
+sudo: required
```
Comment (sonnyp) [inline .travis.yml]: why?

**Comment by sonnyp** [comment]
@c4milo @julianduque out of curiosity; why should it use GCC 4.8?

File: .travis.yml (hunk 1 above)
Comment (julianduque) [inline .travis.yml]: to install g++-4.8, it's required due newer versions of V8

Comment (sonnyp) [inline .travis.yml]: sudo is not needed for installation of packages via apt addon

Comment (julianduque) [inline .travis.yml]: thought it was required, I rollback the value back to false then

**Comment by c4milo** [comment]
@sonnyp the short version is that newer versions of node requires it and travis comes with an older version of g++.

**Comment by sonnyp** [comment]
thanks a lot; released/published as ```2.3.9```

**Comment by c4milo** [comment]
\o/, thanks for reviewing @sonnyp!

**Pull request merged**